### PR TITLE
Add new task :global_deploy

### PR DESCRIPTION
**Pull request by smshahed**
Add a new task `:global_deploy` under the `kubernetes` namespace that would allow to deploy non name-spaced resources to kubernetes, i.e. - ClusterRole, ClusterRoleBinding etc. The task would use the `krane global-deploy` command to apply the resources.

Please note, the `--selector` option is required for `krane global-deploy`.

### CODE REVIEW

File: lib/mina/kubernetes.rb [hunk 1]
```diff
@@ -150,3 +157,21 @@ def apply_kubernetes_resources(options)
...
+
+def apply_global_kubernetes_resources(options)
+  run :local do
+    comment "Applying all global Kubernetes resources..."
+
+    proxy_env = "HTTPS_PROXY=#{fetch(:proxy)}" if fetch(:proxy)
+    filepaths = options&.[](:filepaths) || "config/deploy/#{fetch(:stage)}"
+
+    render_cmd = "#{proxy_env} krane render --bindings=image_repo=#{fetch(:image_repo)},image_tag=#{fetch(:image_tag)},nzamespace=#{fetch(:namespace)} --current_sha #{fetch(:image_tag)} -f #{filepaths}"
+    deploy_cmd = "#{proxy_env} krane global-deploy #{fetch(:kubernetes_context)} --selector app=broadcast-capturer  --stdin "
```
Comment (timflapper) [inline lib/mina/kubernetes.rb]: If we need to include the app parameter it should be configurable: `app=broadcast-capturer`

Comment (smshahed) [reply lib/mina/kubernetes.rb]: ohh, missed that one! thanks for pointing out.

Comment (smshahed) [reply lib/mina/kubernetes.rb]: Is there any other param apart from `app` that we could use as a `selector`?

If we're okay with `app`, I'm thinking to do it like this - 

```
deploy_cmd = "#{proxy_env} krane global-deploy #{fetch(:kubernetes_context)} --selector app=#{fetch(:application_name)}  --stdin "
```
`:application_name` is set in the `config/deploy.rb` for broadcast-capturer. Not sure what would be ideal for other repos.

Comment (timflapper) [reply lib/mina/kubernetes.rb]: Do we even need to deploy with the selector?

Comment (timflapper) [reply lib/mina/kubernetes.rb]: If we do, let's make the entire selector value configurable.

Comment (smshahed) [reply lib/mina/kubernetes.rb]: Yes, it's required to deploy with selector.

File: lib/mina/kubernetes.rb [hunk 2]
```diff
@@ -18,6 +18,13 @@
     apply_kubernetes_resources(args[:options])
   end
 
+  task :global_deploy, [:options] do |task, args|
+    set_tag_from_branch_commit unless fetch(:image_tag)
+    wait_until_image_ready(fetch(:image_tag)) unless fetch(:skip_image_ready_check)
+    create_namespace_on_cluster
```
Comment (asabourin) [inline lib/mina/kubernetes.rb]: Is this one still needed if it's deploying to the global namespace?

Comment (smshahed) [reply lib/mina/kubernetes.rb]: i wouldn't think so, thanks!

**Review by timflapper — APPROVED** [review]

**Review by asabourin — APPROVED** [review]

**Pull request merged**